consensus/src/experimental/tests/commit_phase_tests.rs

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::experimental::commit_phase::CommitPhase;
+use diem_metrics::IntGauge;
+use consensus_types::executed_block::ExecutedBlock;
+use diem_types::ledger_info::{LedgerInfoWithSignatures, LedgerInfo};
+use crate::test_utils::{consensus_runtime, MockStateComputer, MockTransactionManager, MockStorage};
+use crate::network_tests::{NetworkPlayground, TwinId};
+use std::sync::Arc;
+use std::sync::atomic::AtomicU64;
+use channel::message_queues::QueueStyle;
+use diem_types::epoch_state::EpochState;
+use network::peer_manager::{PeerManagerRequestSender, ConnectionRequestSender, conn_notifs_channel};
+use crate::network::NetworkSender;
+use futures::channel::mpsc;
+use crate::util::time_service::ClockTimeService;
+use crate::block_storage::BlockStore;
+use crate::liveness::proposal_generator::ProposalGenerator;
+use crate::metrics_safety_rules::MetricsSafetyRules;
+use crate::network_interface::{ConsensusMsg, ConsensusNetworkSender};
+use channel::{diem_channel, Sender};
+use futures::{stream::select, SinkExt, Stream, StreamExt};
+use diem_infallible::Mutex;
+use futures::channel::mpsc::UnboundedReceiver;
+use consensus_types::common::Payload;
+use diem_types::validator_verifier::random_validator_verifier;
+use safety_rules::{PersistentSafetyStorage, SafetyRulesManager, SafetyRules, TSafetyRules};
+use diem_crypto::{ed25519::{Ed25519PrivateKey, Ed25519Signature}, Uniform};
+use consensus_types::block::Block;
+use diem_types::validator_signer::ValidatorSigner;
+use consensus_types::block::block_test_utils::certificate_for_genesis;
+use crate::state_replication::StateComputer;
+use diem_crypto::hash::ACCUMULATOR_PLACEHOLDER_HASH;
+use std::collections::BTreeMap;
+use diem_types::account_address::AccountAddress;
+use diem_secure_storage::Storage;
+use diem_types::waypoint::Waypoint;
+use futures::executor::block_on;
+use network::{
+    protocols::network::{Event, NewNetworkEvents, NewNetworkSender},
+};
+use std::time::Duration;
+
+
+fn prepare_commit_phase(num_nodes: usize) -> (
+    Sender<(Vec<ExecutedBlock>, LedgerInfoWithSignatures)>,
+    Sender<ConsensusMsg>,
+    UnboundedReceiver<LedgerInfoWithSignatures>,
+    UnboundedReceiver<Payload>,
+    ProposalGenerator,
+    Arc<Mutex<MetricsSafetyRules>>,
+    Vec<ValidatorSigner>,
+    Arc<MockStateComputer>,
+){
+    assert!(num_nodes > 0);
```

Comment:
we don't need this to be parameter, this can always be 1

Reply:
resolved